src/Comms.cpp

```diff
@@ -22,9 +41,59 @@ namespace Comms {
     }
 
     void emitPacket(Packet *packet) {
-        Serial.print(packet->data[0]);
-        Serial.print(packet->data[1]);
-        Serial.print(packet->data[2]);
-        Serial.println(packet->data[3]);
+        //calculate and append checksum to struct
+        uint16_t checksum = computePacketChecksum(packet);
+        uint8_t *ptr = (uint8_t *)&checksum;
+        packet->checksum[0] = ptr[0];
+        packet->checksum[1] = ptr[1];
+
+        //Send over serial
+        Serial.write(packet->id);
+        Serial.write(packet->len);
+        Serial.write(packet->checksum, 2);
+        Serial.write(packet->data, packet->len);
+        Serial.write('\n');
+
+        //Send over ethernet
+        Udp.beginPacket(Udp.remoteIP(), Udp.remotePort());
+        Udp.write(packet->id);
+        Udp.write(packet->len);
+        Udp.write(packet->checksum, 2);
+        Udp.write(packet->data, packet->len);
+        Udp.endPacket();
+    }
+
+    /**
+     * @brief Computes and returns a 16 byte checksum of a packet
+     */
+    uint16_t computePacketChecksum(Packet *packet) {
+        uint8_t vals[packet->len + 2];
+        vals[0] = packet->id;
+        vals[1] = packet-> len;
+        for (int i = 0; i < packet->len; i++) {
+            vals[i + 2] = packet->data[i];
+        }
+        return fletcher16(vals, packet->len + 2);
+    }
```

Comment:
This seems like a somewhat inefficient way to do the checksum. I'd actually just modify the fletcher16 function directly. If you just set `sum1` equal to `packet->id` and `sum2` equal to `(packet-> id + packet-> len) % 256`, its the equivalent of this function.